### PR TITLE
Team page

### PR DESCRIPTION
**Pull request by petya0111**
This is not a final pull request but a proposal for a better story of the project we are working on :)

Please add in comments your ideas for project history.

- An idea for the detail page of the member when clicking on the card (summary experience)- on popup  
Technologies information and a short description profile of the person
- [x] popup added
- **Ideas for a better-styled ad for inspirational Discord server** 
- accordion history of the project
- [x] accordion added

### CODE REVIEW

File: client/src/components/Header/Team/TeamHistory.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+        <Box className="body">
+            <Box className="banner"  >
+                <Typography className="group-info" variant="h2">
+                    Team Exordium Eshop Project History
+                </Typography>
+                <Typography className="group-info" variant="h6">
+                This is an all purpose e-shop practice project, maintain by a group of highly motivated young developers.
+                </Typography>
+                <Team></Team>
```
Comment (dplamenov) [inline client/src/components/Header/Team/TeamHistory.js]: Use self-closing tag.

File: client/src/components/Header/Header.js [hunk 2]
```diff
@@ -40,6 +41,13 @@ const Header = () => {
               </Typography>
             </Typography>
           </Grid>
+          <Grid item={true} xs={12} sm={2} md={0.5}>
+            <Link to='/team' style={textDecorationObject}>
+              <Button className='header-buttons'>
+                <CodeIcon></CodeIcon>
```
Comment (dplamenov) [inline client/src/components/Header/Header.js]: Use self-closing tag.

File: client/src/components/Header/Header.js [hunk 3]
```diff
@@ -75,46 +83,46 @@ const Header = () => {
 
           <Grid item={true} xs={12} sm={2} md={1.5}>
             {
-              languages.map((item,index) => {
+              languages.map((item, index) => {
```
Comment (dplamenov) [inline client/src/components/Header/Header.js]: Rename var to `language` instead `item`

File: client/src/components/Header/Team/Team.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        {
+            name: 'Jordan Radushev',
+            position: "Backend Developer",
+            linkedin: "https://www.linkedin.com/in/jordan-radushev-76665b1a4/",
+            github: "https://github.com/JordanRad",
+            imgsrc: "https://avatars.githubusercontent.com/u/63496500?v=4"
+        },
+        {
+            name: 'Ivaylo Slavchev',
+            linkedin: "https://github.com/IvayloSlavchev",
```
Comment (Arabadzhiew) [inline client/src/components/Header/Team/Team.js]: Here we should have the url for Ivaylo's LinkedIn.

**Review by Arabadzhiew — CHANGES_REQUESTED** [review]

**Comment by Arabadzhiew** [comment]
@petya0111 I think for the image sources, we should host our own ones, or at least use a more reliable provider. Because, let's say that someone changes their profile picture on GitHub, then the url that we are relying on might no longer work. 


@Arabadzhiew  Now images are hosted ✓

File: client/src/components/Header/Team/Team.js [hunk 5]
```diff
@@ -16,7 +18,14 @@ const Team = () => {
...
-            imgsrc: "https://media-exp1.licdn.com/dms/image/C4E03AQH3Umcu8klQJQ/profile-displayphoto-shrink_400_400/0/1630830179729?e=1664409600&v=beta&t=nJ4Jb76ZamQ5E9DV5tmwyYvZiv3X0AnjLYulpHh5nUs"
+            imgsrc: "https://media-exp1.licdn.com/dms/image/D4D35AQE4y26QIx7SVA/profile-framedphoto-shrink_400_400/0/1658997884989?e=1659729600&v=beta&t=qbgu_C2yIoXwXz2ekWZlF9PZoY_gQRHj7bC9o97BkMA"
+        },
+        {
+            name: 'Mehmed Dukov',
+            linkedin: "https://www.linkedin.com/in/mehmed-mehmedov-dukov-07a18a225/",
+            position: "Fronend Developer",
+            github: "https://github.com/Medikko",
+            imgsrc: "https://media-exp1.licdn.com/dms/image/C4D03AQHv6FsZ7o8Yng/profile-displayphoto-shrink_400_400/0/1658952648967?e=1664409600&v=beta&t=CQCams4dENjaVC7XAbiqgZJffpjPjBn9ugUSbnDdb2s"
         },
```
Comment (IvayloSlavchev) [inline client/src/components/Header/Team/Team.js:41]: @Medikko is a Back-End developer

File: client/src/components/Header/Team/Team.js [hunk 6]
```diff
@@ -25,46 +34,48 @@ const Team = () => {
             github: "https://github.com/Arabadzhiew",
             imgsrc: "https://avatars.githubusercontent.com/u/76776739?v=4"
         },
```
Comment (IvayloSlavchev) [inline client/src/components/Header/Team/Team.js:36]: @Arabadzhiew is a Full-Stack developer and also you should provide a link to his LinkedIn account.

**Review by IvayloSlavchev — CHANGES_REQUESTED** [review]

**Review by JordanRad — APPROVED** [review]

File: client/src/components/Header/Team/Team.js (hunk 5 above)
Comment (JordanRad) [reply client/src/components/Header/Team/Team.js:41]: I am also a full-stack developer, but that's a minor detail at this point

File: client/src/components/Header/Team/TeamHistory.js (hunk 1 above)
Comment (petya0111) [reply client/src/components/Header/Team/TeamHistory.js]: Fixed Team self closing tag ✓ @dplamenov

**Review by IvayloSlavchev — APPROVED** [review]
Good job, @petya0111!

File: client/src/components/Header/Header.js (hunk 2 above)
Comment (petya0111) [reply client/src/components/Header/Header.js]: Fixed CodeIcon self-closing tag ✓ @dplamenov

File: client/src/components/Header/Header.js (hunk 3 above)
Comment (petya0111) [reply client/src/components/Header/Header.js]: Fixed languageItem variable name ✓ @dplamenov

File: client/src/components/Header/Team/Team.js (hunk 4 above)
Comment (petya0111) [reply client/src/components/Header/Team/Team.js]: Fixed Ivaylo's Linkedin ✓ @Arabadzhiew

File: client/src/components/Header/Team/Team.js (hunk 5 above)
Comment (petya0111) [reply client/src/components/Header/Team/Team.js:41]: Fixed roles ✓ @IvayloSlavchev  @JordanRad

**Review by Arabadzhiew — APPROVED** [review]
Good work!

File: client/src/components/Header/Team/TeamHistory.js (hunk 1 above)
Comment (petya0111) [reply client/src/components/Header/Team/TeamHistory.js]: @dplamenov  I am curious to know what is your nickname in discord :)

Comment (dplamenov) [reply client/src/components/Header/Team/TeamHistory.js]: DimitarDimitrov#4430

**Review by fbozhkov — APPROVED** [review]
Great work Petya! We just need to find an elegant way to incorporate the team button.

**Review by IvayloSlavchev — APPROVED** [review]
I think we can merge it. Btw we have to do some styling at first place, but I think that I can handle some tasks. For now everything seems good. Great job, @petya0111!

File: server/controllers/Users.js [hunk 7]
```diff
@@ -20,6 +20,12 @@ usersController.post('/registration', async (req, res) => {
     await UserService.registerUser(userData)
     res.json(userData)
 })
+usersController.get('/check-username/{username}', async (req, res) => {
+    console.log("int the endpoint check username")
```
Comment (JordanRad) [inline server/controllers/Users.js:24]: please, remove the console.log

File: server/controllers/Users.js [hunk 8]
```diff
@@ -20,6 +20,12 @@ usersController.post('/registration', async (req, res) => {
     await UserService.registerUser(userData)
     res.json(userData)
 })
+usersController.get('/check-username/{username}', async (req, res) => {
```
Comment (JordanRad) [inline server/controllers/Users.js:23]:
```suggestion
usersController.get('/username/{username}', async (req, res) => {
```
You might just proceed with /username/{username}. Implicitly, a GET request checks if a given resource is present or not

Comment (JordanRad) [reply server/controllers/Users.js:23]: Avoid using verbs within an API path. Refer to this: https://restfulapi.net/resource-naming/

File: server/services/UserService.js [hunk 9]
```diff
@@ -142,4 +142,27 @@ export default class UserService {
...
+            }).catch(error => { })
+            if (userUsername) {
+                return true;
+            }
+        } else {
+            return true;
+        }
+
+        return false;
+    }
```
Comment (JordanRad) [inline server/services/UserService.js:167]: You can simply check only if a user with a given username exists. You are not expected to face a situation in which a user has username as property and empty email.

**Review by JordanRad — CHANGES_REQUESTED** [review]
Great work! I've made some short comments regarding minor details

File: server/controllers/Users.js (hunk 7 above)
Comment (petya0111) [reply server/controllers/Users.js:24]: this is out of scope from team page

File: server/controllers/Users.js (hunk 8 above)
Comment (petya0111) [reply server/controllers/Users.js:23]: this is out of scope from team page

File: server/services/UserService.js (hunk 9 above)
Comment (petya0111) [reply server/services/UserService.js:167]: this is out of scope from team page

**Pull request merged**